lightning/src/routing/router.rs

```diff
@@ -30,47 +30,60 @@ use crate::prelude::*;
 use crate::sync::Mutex;
 use alloc::collections::BinaryHeap;
 use core::cmp;
-use core::ops::Deref;
+use core::ops::{Deref, DerefMut};
 
 /// A [`Router`] implemented using [`find_route`].
 pub struct DefaultRouter<G: Deref<Target = NetworkGraph<L>>, L: Deref, S: Deref> where
 	L::Target: Logger,
 	S::Target: for <'a> LockableScore<'a>,
+	S::Target: Score,
 {
 	network_graph: G,
 	logger: L,
 	random_seed_bytes: Mutex<[u8; 32]>,
-	scorer: S
+	scorer: S,
+	score_params: <S::Target as Score>::ScoreParams
```

Comment:
i tried changing this to `<<<S::Target as LockableScore<'a>::Locked> as Score>::ScoreParams` but that required `'a` in the generics parameters because `for<'a> LockableScore<'a>` isnt a declaration of `'a` that is usable in this line.